src/Avalonia.Controls/ProgressBar.cs

```diff
@@ -106,9 +107,19 @@ public double Container2AnimationEndPosition
         public static readonly StyledProperty<bool> ShowProgressTextProperty =
             AvaloniaProperty.Register<ProgressBar, bool>(nameof(ShowProgressText));
 
+        public static readonly StyledProperty<string> ProgressTextFormatProperty =
+            AvaloniaProperty.Register<ProgressBar, string>(nameof(ProgressTextFormat), "{0:0}%");
+        
         public static readonly StyledProperty<Orientation> OrientationProperty =
             AvaloniaProperty.Register<ProgressBar, Orientation>(nameof(Orientation), Orientation.Horizontal);
 
+        public static readonly DirectProperty<ProgressBar, double> PercentageProperty =
+            AvaloniaProperty.RegisterDirect<ProgressBar, double>(
+                nameof(Percentage),
+                o => o.Percentage,
+                (o, v) => o.Percentage = v,
```

Comment:
I think this setter shouldn't be defined (i.e. leave the default `null` value), or it's not really read-only, as it's possible to do `obj.SetValue(PercentageProperty, value)`.